emcc.py

```diff
@@ -3091,16 +3106,28 @@ def run_closure_compiler():
   # this will also remove debug info if we only kept it around in the intermediate invocations.
   # note that if we aren't emitting a binary (like in wasm2js) then we don't
   # have anything to do here.
-  if options.emit_symbol_map and os.path.exists(wasm_target):
-    building.handle_final_wasm_symbols(wasm_file=wasm_target, symbols_file=symbols_file, debug_info=debug_info)
-    save_intermediate_with_wasm('symbolmap', wasm_target)
+  if options.emit_symbol_map:
+    if os.path.exists(wasm_target):
+      building.handle_final_wasm_symbols(wasm_file=wasm_target, symbols_file=symbols_file, debug_info=debug_info)
```

Comment:
Maybe pass `debug_info=intermediate_debug_info` here too for consistency... I guess maybe pre-decrement `intermediate_debug_info` and always pass it in?

If you pre-decrement, and as you say you are guaranteed to hit 0, then by definition the last command that ran will have done the strip.

Reply:
Interesting idea... I think it might work, but I worry that we might need `-g` not only for saving the debug info, but also loading it. If so there might be something to fix on the binaryen side. Let's see what CI says.

Reply:
Nice, looks like no binaryen problems! Tests all passed.